pkg/graph/resolvers/trb_request_status.go

```diff
@@ -117,54 +122,42 @@ func getTRBAdviceLetterStatus(ctx context.Context, store *storage.Store, trbRequ
 }
 
 // GetTRBTaskStatuses retrieves all of the statuses for the steps of a given TRB request's task list
-func GetTRBTaskStatuses(ctx context.Context, store *storage.Store, trbRequest models.TRBRequest) (*models.TRBTaskStatuses, error) {
-	errGroup := new(errgroup.Group)
-
-	var formStatus *models.TRBFormStatus
-	var errForm error
-	errGroup.Go(func() error {
-		formStatus, errForm = getTRBFormStatus(ctx, trbRequest.ID)
-		return errForm
-	})
-
-	var feedbackStatus *models.TRBFeedbackStatus
-	var errFeedback error
-	errGroup.Go(func() error {
-		feedbackStatus, errFeedback = getTRBFeedbackStatus(ctx, trbRequest.ID)
-		return errFeedback
-	})
-
-	var consultPrepStatus *models.TRBConsultPrepStatus
-	var errConsultPrep error
-	errGroup.Go(func() error {
-		consultPrepStatus, errConsultPrep = getTRBConsultPrepStatus(ctx, trbRequest)
-		return errConsultPrep
-	})
-
-	var attendConsultStatus *models.TRBAttendConsultStatus
-	var errAttendConsult error
-	errGroup.Go(func() error {
-		attendConsultStatus, errAttendConsult = getTRBAttendConsultStatus(ctx, trbRequest)
-		return errAttendConsult
-	})
-
-	var adviceLetterStatus *models.TRBAdviceLetterStatus
-	adviceLetterStatusTaskList := models.TRBAdviceLetterStatusTaskListInReview
-	var errAdviceLetter error
-	errGroup.Go(func() error {
-		adviceLetterStatus, errAdviceLetter = getTRBAdviceLetterStatus(ctx, store, trbRequest)
-		if *adviceLetterStatus == models.TRBAdviceLetterStatusCannotStartYet {
-			adviceLetterStatusTaskList = models.TRBAdviceLetterStatusTaskListCannotStartYet
-		} else if *adviceLetterStatus == models.TRBAdviceLetterStatusCompleted {
-			adviceLetterStatusTaskList = models.TRBAdviceLetterStatusTaskListCompleted
-		}
-		return errAdviceLetter
-	})
+func GetTRBTaskStatuses(ctx context.Context, trbRequest models.TRBRequest) (*models.TRBTaskStatuses, error) {
+	formStatus, err := getTRBFormStatus(ctx, trbRequest.ID)
+	if err != nil {
+		return nil, err
+	}
 
-	if err := errGroup.Wait(); err != nil {
+	feedbackStatus, err := getTRBFeedbackStatus(ctx, trbRequest.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	consultPrepStatus, err := getTRBConsultPrepStatus(ctx, trbRequest)
+	if err != nil {
 		return nil, err
 	}
 
+	attendConsultStatus, err := getTRBAttendConsultStatus(ctx, trbRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	adviceLetterStatus, err := getTRBAdviceLetterStatus(ctx, trbRequest)
+	if err != nil {
+		return nil, err
+	}
+	if adviceLetterStatus == nil {
```

Comment:
should we nil-check all of these? i see they are all dereferenced below

also a curiosity - why remove to goroutines?

Reply:
I was thinking that opening a zillion channels is probably worse than just writing it synchronously and letting the dataloaders return cached data, but I might be wrong there. @ClayBenson94 do you have any thoughts?

I missed the dereference! I will add checks.

Reply:
Nil checks added here: [ed83b18](https://github.com/CMS-Enterprise/easi-app/pull/2702/commits/ed83b181ce634ed6af84b34a3115131d6b607051)

Reply:
If each synchronous call is behind a dataloader it probably boils down to readability more than anything, since each of those calls will probably already be cached for this request